remove -m$(ARCH)
add versioning to libhl
{CFLAGS} -> (CFLAGS) to not get picked up by SUBST_CMD
remove Linux's 32-/64-bit detection
remove Darwin detection in install target (breaks 'make fake')

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -1,6 +1,4 @@
 
-LBITS := $(shell getconf LONG_BIT)
-MARCH ?= $(LBITS)
 PREFIX ?= /usr/local
 INSTALL_DIR ?= $(PREFIX)
 INSTALL_BIN_DIR ?= $(PREFIX)/bin
@@ -9,12 +7,12 @@ INSTALL_INCLUDE_DIR ?= $(PREFIX)/include
 
 LIBS=fmt sdl ssl openal ui uv mysql
 
-CFLAGS = -Wall -O3 -I src -msse2 -mfpmath=sse -std=c11 -I include -I include/pcre -I include/mikktspace -I include/minimp3 -D LIBHL_EXPORTS
-LFLAGS = -L. -lhl
+CFLAGS = ${CFLAGS} -I src -I${LOCALBASE}/include -I${X11BASE}/include -std=c11 -I include -I include/pcre -I include/mikktspace -I include/minimp3 -D LIBHL_EXPORTS
+LFLAGS = -L. -L${LOCALBASE}/lib -lhl
 EXTRA_LFLAGS ?=
-LIBFLAGS =
-HLFLAGS = -ldl
-LIBEXT = so
+LIBFLAGS = -L${LOCALBASE}/lib -L${X11BASE}/lib
+HLFLAGS =
+LIBEXT = so.${LIBhl_VERSION}
 LIBTURBOJPEG = -lturbojpeg
 
 PCRE = include/pcre/pcre_chartables.o include/pcre/pcre_compile.o include/pcre/pcre_dfa_exec.o \
@@ -88,16 +86,9 @@ LIB += ${HL_DEBUG}
 else
 
 # Linux
-CFLAGS += -m$(MARCH) -fPIC -pthread -fno-omit-frame-pointer
+CFLAGS += -fPIC -pthread -fno-omit-frame-pointer
 LFLAGS += -lm -Wl,-rpath,. -Wl,--export-dynamic -Wl,--no-undefined
 
-ifeq ($(MARCH),32)
-CFLAGS += -I /usr/include/i386-linux-gnu
-LIBFLAGS += -L/opt/libjpeg-turbo/lib
-else
-LIBFLAGS += -L/opt/libjpeg-turbo/lib64
-endif
-
 LIBOPENAL = -lopenal
 RELEASE_NAME = linux
 
@@ -115,7 +106,6 @@ endif
 all: libhl hl libs
 
 install:
-	$(UNAME)==Darwin && make uninstall
 	mkdir -p $(INSTALL_BIN_DIR)
 	cp hl $(INSTALL_BIN_DIR)
 	mkdir -p $(INSTALL_LIB_DIR)
@@ -131,7 +121,7 @@ uninstall:
 libs: $(LIBS)
 
 libhl: ${LIB}
-	${CC} -o libhl.$(LIBEXT) -m${MARCH} ${LIBFLAGS} -shared ${LIB} -lpthread -lm
+	${CC} -o libhl.$(LIBEXT) ${LIBFLAGS} -shared ${LIB} -lpthread -lm
 
 hlc: ${BOOT}
 	${CC} ${CFLAGS} -o hlc ${BOOT} ${LFLAGS} ${EXTRA_LFLAGS}
@@ -221,7 +211,7 @@ codesign_osx:
 .SUFFIXES : .c .o
 
 .c.o :
-	${CC} ${CFLAGS} -o $@ -c $<
+	${CC} $(CFLAGS) -o $@ -c $<
 
 clean_o:
 	rm -f ${STD} ${BOOT} ${RUNTIME} ${PCRE} ${HL} ${FMT} ${SDL} ${SSL} ${OPENAL} ${UI} ${UV} ${HL_DEBUG}
